finetune/lora.py

```diff
@@ -95,36 +89,64 @@ def setup(
 
     logger = CSVLogger(out_dir.parent, out_dir.name, flush_logs_every_n_steps=log_interval)
     fabric = L.Fabric(devices=devices, strategy=strategy, precision=precision, loggers=logger, plugins=plugins)
-    fabric.print(hparams)
-    fabric.launch(main, data_dir, checkpoint_dir, out_dir)
 
+    if not any((lora_query, lora_key, lora_value, lora_projection, lora_mlp, lora_head)):
+        fabric.print("Warning: all LoRA layers are disabled!")
+    fabric.launch(
+        main,
+        devices,
+        Config.from_name(
+            name=checkpoint_dir.name,
+            r=lora_r,
+            alpha=lora_alpha,
+            dropout=lora_dropout,
+            to_query=lora_query,
+            to_key=lora_key,
+            to_value=lora_value,
+            to_projection=lora_projection,
+            to_mlp=lora_mlp,
+            to_head=lora_head,
+        ),
+        IOArgs(data_dir, checkpoint_dir, out_dir),
+        TrainArgs(
```

Comment:
Is this granularity for practical reasons? Would we consider merging train args, eval args and optimization args? They are all train args IMO. In other words, when would I want to only use optimizer args?

Reply:
I agree, let’s aggregate where we can. Less cognitive overhead.

Reply:
For instance, in a sweep over lr you'd only need to change one small config

Ill merge this as is and when all the scripts have been ported, revisit this choice.